### PR TITLE
add basic descriptions of tool specs and api, arch and internal api

### PR DESCRIPTION
**Pull request by emanueldima**

### CODE REVIEW

**Comment by emanueldima** [comment]
This PR partially solves https://github.com/clarin-eric/switchboard/issues/3

File: documentation/InternalAPI.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+## GET /api/tools?mediatype=[MEDIATYPE]&language=[LANGUAGE]&deployment=[DEPLOYMENT]
+
+Returns a JSON document with descriptions of all tools registered with the Switchboard that match the designated values supplied by the arguments (any order permitted):
+
+- `mediatype`: find tools that can process the specified mediatype. The a mediatype identifier follows the [classification of IANA.org](https://www.iana.org/assignments/media-types/media-types.xhtml).
+
+- `language`: find tools that can process the specified language. The language identifier must use the [ISO 639-3](https://iso639-3.sil.org/code_tables/639/data) standard.
+
+- `deployment`: specifies the inclusion of tools (and web services) that are deployed in `production` or `development` instances. If absent, all tools are returned.
```
Comment (andmor-) [inline documentation/InternalAPI.md:19]: We will have to review this one in line with switchboard [#47](https://github.com/clarin-eric/switchboard/issues/47)

File: documentation/ToolDescriptionSpec.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+Tool Description Specification, v1.0
+-------------------------------------
+
+A Tool is specified in Switchboard by a JSON file, with the following entries:
+
+* `task`: A short description of a tool's task, e.g. "Constituency Parsing".
+
+* `deployment`: The status of its instance, should be either "production", or "development".
+
+* `softwareType`: Can be "qualitative" or "quantitative".
```
Comment (andmor-) [inline documentation/ToolDescriptionSpec.md:10]: As discussed this one will be dropped

**Review by andmor- — APPROVED** [review]
Two comments due to the latest discussions, but in general I think we should merge this and modify these details after the changes are implemented in the Switchboard.

**Pull request merged**